src/main/java/io/openliberty/tools/intellij/lsp4mp4ij/psi/internal/core/java/corrections/JavaDiagnosticsDefinition.java

```diff
@@ -50,49 +51,48 @@ public final class JavaDiagnosticsDefinition extends BaseKeyedLazyInstance<IJava
 
     @Override
     public boolean isAdaptedForDiagnostics(JavaDiagnosticsContext context) {
-        try {
-            return getInstance().isAdaptedForDiagnostics(context);
-        } catch (IndexNotReadyException | ProcessCanceledException | CancellationException e) {
-            throw e;
-        } catch (Exception e) {
-            LOGGER.log(Level.WARNING, "Error while calling isAdaptedForDiagnostics", e);
-            return false;
-        }
+        return ExceptionUtil.executeWithExceptionHandling(
+                () -> getInstance().isAdaptedForDiagnostics(context),
+                () -> false,  // Fallback value in case of exception
+                e -> LOGGER.log(Level.WARNING, "Error while calling isAdaptedForDiagnostics", e)
+        );
     }
 
     @Override
     public void beginDiagnostics(JavaDiagnosticsContext context) {
-        try {
-            getInstance().beginDiagnostics(context);
-        } catch (IndexNotReadyException | ProcessCanceledException | CancellationException e) {
-            throw e;
-        } catch (Exception e) {
-            LOGGER.log(Level.WARNING, "Error while calling beginDiagnostics", e);
-        }
+        ExceptionUtil.executeWithExceptionHandling(
+            () -> {
+                getInstance().beginDiagnostics(context);
+                return true;
+            },
+                null,
```

Comment:
`NullPointerException`? See my previous comment.

Reply:
I made the change like this 

```
@Override
    public void beginDiagnostics(JavaDiagnosticsContext context) {
        ExceptionUtil.executeWithExceptionHandling(
            () -> {
                getInstance().beginDiagnostics(context);
                return true;
            },
            e -> {
                LOGGER.log(Level.WARNING, "Error while calling beginDiagnostics", e);
                return null;
            }
        );
    }
```

This won't give `NullPointerException`.